packages/cli/src/constructs/alert-escalation-policy.ts

```diff
@@ -1,33 +1,113 @@
+/** Types of alert escalation strategies */
 // eslint-disable-next-line no-restricted-syntax
 enum AlertEscalationType {
   RUN = 'RUN_BASED',
   TIME = 'TIME_BASED'
 }
 
+/**
+ * Configuration for alert reminders.
+ * Defines how often to send reminder notifications after initial alert.
+ */
 export type Reminders = {
+  /** 
+   * Number of reminder notifications to send (0 to disable).
+   * @defaultValue 0
+   * @minimum 0
+   * @maximum 5
+   * @example 2  // Send 2 reminder notifications
+   */
   amount?: number,
+  
+  /** 
+   * Interval between reminder notifications in minutes.
+   * @defaultValue 5
+   * @enum [5, 10, 15, 30]
+   */
   interval?: number
 }
 
+/**
+ * Configuration for parallel run failure threshold.
+ * Determines when to alert based on percentage of failed parallel runs.
+ */
 export type ParallelRunFailureThreshold = {
+  /** 
+   * Whether parallel run failure threshold is enabled.
+   * @defaultValue false
+   */
   enabled?: boolean,
+  
+  /** 
+   * Percentage of runs that must fail to trigger alert.
+   * @defaultValue 10
+   * @minimum 10
+   * @maximum 100
+   * @multipleOf 10
+   * @example 30  // Alert when 30% of parallel runs fail
+   */
   percentage?: number,
 }
 
+/**
+ * Configuration for alert escalation policies.
+ * Defines when and how to escalate alerts based on check failures.
+ */
 export interface AlertEscalation {
+  /** The type of escalation strategy to use */
   escalationType?: AlertEscalationType,
+  /** Configuration for run-based escalation */
   runBasedEscalation?: {
+    /** 
+     * Number of consecutive failed runs before escalating.
+     * @defaultValue 1
```

Comment:
I don't actually see a default value for this at the backend level, or at least at the schema level, when `runBasedEscalation` is not specified, or set to an empty object, or `alertSettings` is left out entirely. It is true however that if the user uses `AlertEscalationBuilder`, then the default value is indeed `1`. But if the user manually creates the `alertEscalationPolicy`, I'm not sure what the effective default will end up being. This is probably a bug that should be fixed in the backend tbh.